lesson01/assignments/scripts.js

```diff
@@ -1,26 +1,124 @@
 // jQuery Doc Ready
 $(function () {
-    // Set up some data and variables
-    const data = [1, 2, 3, 4],
-        input = $('.input'),
-        submit = $('.submit'),
-        resultSum = $('.sum');
-    // Click event on the submit input
-    submit.on('click', function () {
-        // Grab the input value and parse it into a number
-        const val = input.val();
-        const num = parseInt(val);
-        // Verify the user gave us a real number
-        if (!isNaN(num)) {
-            // Add our new number to the data array
-            data.push(num);
-            // Use reduce to convert our array of numbers into a single value - the sum of all of the numbers
-            const sum = data.reduce((acc, curr) => acc + curr);
-            // Add our new total to the page
-            resultSum.text(sum);
-            // Reset the input value to an empty string
-            input.val('');
-        }
-    });
-});
+	// Set up some data and variables
+	const data = [18, 2, 3, 4],
+		input = $('.input'),
+		submit = $('.submit'),
+		resultSum = $('.sum'), //reduce example
+		result1 = $('.power'), //map
+		result2 = $('.power2'), //foreach
+		result3 = $('.smallmore3'), //find sort
+		result4 = $('.even'), //filter
+		result5 = $('.multiply'), //reduce
+		result6 = $('.lessthan3'), //filter length
+		result7 = $('.fib');
 
+
+	// Click event on the submit input
+	submit.on('click', function () {
```

Comment:
Rather than a single submit event handler that listens for events on any `.submit` item, separate handlers for each submit button would result in code that's easier to understand and maintain. The current strategy of getting correct values from sibling elements, relies on placement of elements, which makes it brittle.

Example HTML for fibonacci section (added class "fibonacci" to containing div): 
```
         <div class='b7 fibonacci'>
            <!-- Input -->
            <!--return fibonnacci number for index given (0 indexed)-->
            <input type="text" class="input7 input">
            <!-- Submit for Input -->
            <input  id='button7' class="submit" type="submit">
            <!-- Result markup with hook for answer -->
            <p class="markup"> <span class="fib"></span></p>
         </div>
```

Example listener code: 
```
const fibSubmit = $('.fibonacci > .submit')
fibSubmit.on('click', function () { 
   const val = $('.fibonacci > .input')
  // math
  const output = $('.fibonacci > .markup')
  output.text('the result is...')
 }
```